### DISCUSSION TITLE
Unable to spin up cluster with Loadbalancer listener

### DISCUSSION THREAD

**Discussion by spoorthyredi2** [strimzi/strimzi-kafka-operator#7567] [Q&A] [answered]
Hi everyone.

I am trying to set up 3-broker Kafka cluster using Strimzi Helm chart operator and Kafka persistent yaml on GKE. 
Plain Listener works as expected. 

Whenever I configure LoadBalancer type listener, (with tls=false and tls=true), both ways,   after the zookeeper pods are all running, the Kafka pods are not created at all.   

When I configure listener with NodePort type listener , with tls=false,  Kafka pods come up , cluster looks as expected... but  I am not able to connect to cluster through the right IP and nodeport.   ... and I see below error in producer / consumer apps.

when I try to send messages , "Couldnt connect to broker etc" . and When I abort the producer app -- " Aborted while sending is in progress ..."

I followed the NodePort documentation to get the bootstrap ip and nodeport, but that itself isnt working.
I suspect something with my cluster setup.
Please help

**Comment by scholzj** [comment, marked as answer]
> Whenever I configure LoadBalancer type listener, (with tls=false and tls=true), both ways, after the zookeeper pods are all running, the Kafka pods are not created at all.

Try to do `kubectl get service -o wide` and share the output. I expect that the loadbalancer service will be there as `<pending>` => that means they were not provisioned by your GKE. That could be for example because you run out of some limit for maximal number of load balancers, or you are not allowed to have load balancers, or you can have only internal loadbalancers etc. It really depends on your Kube cluster and your infrastructure.

> When I configure listener with NodePort type listener , with tls=false, Kafka pods come up , cluster looks as expected... but I am not able to connect to cluster through the right IP and nodeport. ... and I see below error in producer / consumer apps.

Again, this might depend on your infrastructure. You will likely need to allow access tot he node ports in some firewalls for example in your Google Cloud account.

**Reply by spoorthyredi2** [reply]
@scholzj 
Thanks for your reply. You were right.  My lb services are in pending state. 

```
[root@osboxes strimziKAFKA]# kubectl get svc -n strimzi-kafka
W1103 07:01:31.293586    7692 gcp.go:119] WARNING: the gcp auth plugin is deprecated in v1.22+, unavailable in v1.26+; use gcloud instead.
To learn more, consult https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
NAME                                           TYPE           CLUSTER-IP    EXTERNAL-IP   PORT(S)                      AGE
my-kafka-cluster4-kafka-bootstrap              ClusterIP      10.48.0.146   <none>        9091/TCP,9097/TCP            40s
my-kafka-cluster4-kafka-brokers                ClusterIP      None          <none>        9090/TCP,9091/TCP,9097/TCP   40s
my-kafka-cluster4-kafka-plainloadb-0           LoadBalancer   10.48.14.8    <pending>     9099:30726/TCP               40s
my-kafka-cluster4-kafka-plainloadb-1           LoadBalancer   10.48.12.0    <pending>     9099:31494/TCP               40s
my-kafka-cluster4-kafka-plainloadb-2           LoadBalancer   10.48.0.41    <pending>     9099:30629/TCP               40s
my-kafka-cluster4-kafka-plainloadb-bootstrap   LoadBalancer   10.48.7.103   <pending>     9099:31691/TCP               40s
my-kafka-cluster4-zookeeper-client             ClusterIP      10.48.9.171   <none>        2181/TCP                     80s
my-kafka-cluster4-zookeeper-nodes              ClusterIP      None          <none>        2181/TCP,2888/TCP,3888/TCP   80s
```

I see below error on one of them:

Events:
  Type     Reason                  Age                From                Message
  ----     ------                  ----               ----                -------
  Normal   EnsuringLoadBalancer    22s (x4 over 62s)  service-controller  Ensuring load balancer
  Warning  SyncLoadBalancerFailed  20s (x4 over 60s)  service-controller  Error syncing load balancer: failed to ensure load balancer: failed to ensure a static IP for load balancer (aa325adf1e51f49159e76b1d777d62af(strimzi-kafka/my-kafka-cluster4-kafka-plainloadb-0)): error creating gce static IP address: googleapi: Error 403: QUOTA_EXCEEDED - Quota 'STATIC_ADDRESSES' exceeded.  Limit: 8.0 in region us-east1.
[root@osboxes strimziKAFKA]#

**Reply by scholzj** [reply]
Yeah, so I guess you will need to first get the limit increased. I do not use Google Cloud much. In Amazon AWS there is a form for that in the AWS Console to request it and normally they come back in few hours. So I guess itmight work similarly.

**Reply by spoorthyredi2** [reply]
Yes, its the same with GCP . I will get the limit increased and get back right after ... thanks much

**Reply by spoorthyredi2** [reply]
@scholzj 

We have increased limit for static IPs and the pending issue is gone.

However,  I see only one pod kafka pod  being in pending state all the time , inspite of adding resources. 

![image](https://user-images.githubusercontent.com/116352972/199710693-1882352f-a307-4da8-9da2-db94f7bd8d9f.png)

Pod desc : 
  Warning  FailedScheduling   48s   default-scheduler   0/5 nodes are available: 2 Insufficient cpu, 3 node(s) had volume node affinity conflict.
  Normal   NotTriggerScaleUp  46s   cluster-autoscaler  pod didn't trigger scale-up:

**Reply by spoorthyredi2** [reply]
Every time I add a new node, It only goes into the " node(s) had volume node affinity conflict." category. I wonder why

**Reply by scholzj** [reply]
Again, I do not know GKE that well and I do not know what storage you use. The PV/PVC might be for example bound to some availability zone. And in that case it is not enough to have free nodes, but you need to have free nodes in the right zone. BAsically, this message:

```
0/5 nodes are available: 2 Insufficient cpu, 3 node(s) had volume node affinity conflict
```

Says that 2 nodes are in the position to mount the persistent volume but do not have CPU. The other 3 nodes might have the CPU, but they cannot mount the volume (for example because they are in a different zone).

**Reply by spoorthyredi2** [reply]
@scholzj  after fixing the resources issue , all 3 brokers pods are created , however i dont see entity operator Pod... It should automatically create after the broker pods right ..   please can u suggest

![image](https://user-images.githubusercontent.com/116352972/199747379-412c4dc9-9451-4ced-ac57-6be64ed1733f.png)

**Reply by spoorthyredi2** [reply]
@scholzj  Ignore, it came up now and Kafka is ready. Will try to access

**Reply by scholzj** [reply]
I guess you would need to check the logs etc. There is no way to know what the problem is just from the screenshot.

**Reply by spoorthyredi2** [reply]
@scholzj Thankyou so much !! It worked and I am able to test basic messaging access via LB IP and port

**Reply by scholzj** [reply]
Great, glad it works. 👍
